wolfcrypt/src/misc.c

```diff
@@ -131,6 +131,13 @@ WC_STATIC WC_INLINE word32 ByteReverseWord32(word32 value)
     /* 5 instructions with rotate instruction, 9 without */
     return (rotrFixed(value, 8U) & 0xff00ff00) |
            (rotlFixed(value, 8U) & 0x00ff00ff);
+#elif defined(WOLFSSL_ARM)
```

Comment:
Would it fit with WOLFSSL_ARMASM?

Reply:
Thanks Kojo!
I meant to fix this to be turned on for ARM compilation regardless of WOLFSSL_ARMASM.
Fixed it to detect the CPU based on __arm__ and __aarch64__.

Reply:
An user reports that other compiler like Diab VxWorks seems to have direct of inline assembler, and gets error at this line. It maight be good to separate switches for architechture and asm.

Reply:
OK. Added check for __GNUC__ now before using assembly code.